crates/matrix-sdk-common/src/ttl_cache.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2025 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A TTL cache which can be used to time out repeated operations that might
+//! experience intermittent failures.
+
+use std::{borrow::Borrow, collections::HashMap, hash::Hash, time::Duration};
+
+use ruma::time::Instant;
+
+const DEFAULT_LIFETIME: Duration = Duration::from_secs(15 * 60);
```

Comment:
Why only 15 minutes? It seems like with the given issue, the goal is to be able to have the URL with no connection, or a slow connection. The URL is unlikely to change so maybe a week would be fine, like for the server capabilities?

Reply:
No particular reason, just  seemed to be the minimum viable value to remove the flicker from the linked issue.

Let's put it to a day: d114feea80b74196bb72677b5f83fa846083b2f4.